backend/lib/endpoints/posts.js

```diff
@@ -374,42 +374,122 @@ async function routes(app) {
     },
   );
 
-  app.post(
+  // -- Comments
+  const COMMENT_PAGE_SIZE = 5;
+
+  app.get(
     "/:postId/comments",
-    { preValidation: [app.authenticate], schema: addCommentSchema },
+    { preValidation: [app.authenticate], schema: getCommentsSchema },
     async (req) => {
-      const { body, params, userId } = req;
-      const { parentId } = body;
-      const { postId } = params;
+      const { limit, skip } = req.query;
+      const { postId } = req.params;
+      const [commentErr, comments] = await app.to(
+        Comment.aggregate([
+          {
+            $match: {
+              parentId: null,
+              postId: mongoose.Types.ObjectId(postId),
+            },
+          },
+          {
+            $lookup: {
+              as: "children",
+              foreignField: "parentId",
+              from: "comments",
+              localField: "_id",
+            },
+          },
+          {
+            $addFields: {
+              childCount: {
+                $size: { $ifNull: ["$children", []] },
+              },
+            },
+          },
+          {
+            $skip: skip || 0,
+          },
+          {
+            $limit: limit || COMMENT_PAGE_SIZE,
+          },
+        ]),
+      );
+      if (commentErr) {
+        req.log.error("Failed retrieving comments", { commentErr });
+        throw app.httpErrors.internalServerError();
+      }
+
+      return comments;
+    },
+  );
+
+  app.post(
+    "/:postId/comments",
+    { preValidation: [app.authenticate], schema: createCommentSchema },
+    async (req, reply) => {
+      const userId = req.userId;
+      const { parentId } = req.body;
+      const { postId } = req.params;
+      const { body: commentProps } = req;
+
+      const [userErr, user] = await app.to(User.findById(userId));
+      if (userErr) {
+        req.log.error("Failed retrieving user", { userErr });
+        throw app.httpErrors.internalServerError();
+      }
+
+      // Assign postId and parent comment id (if present)
+      commentProps.postId = postId;
       if (parentId) {
-        const parentPost = await Post.findById(parentId);
-        if (!parentPost || parentPost.postId !== postId) {
-          return new httpErrors.BadRequest();
-        }
+        commentProps.parentId = parentId;
       }
-      return new Comment({
-        ...body,
-        authorId: userId,
-        postId,
-      }).save();
+
+      // Creates embedded author document
+      commentProps.author = {
+        id: user.id,
+        location: user.location,
+        name: user.name,
+        type: user.type,
+      };
+
+      // Initial empty likes array
+      commentProps.likes = [];
+
+      const [err, comment] = await app.to(new Comment(commentProps).save());
+
+      if (err) {
+        req.log.error("Failed creating comment", { err });
+        throw app.httpErrors.internalServerError();
+      }
+
+      reply.code(201);
+      return comment;
     },
   );
 
-  app.put(
+  app.patch(
```

Comment:
Looking at the designs, I don't think there are options to edit or delete a comment ? I've asked the product team to clarify about this.

Reply:
Good point, from what I've seen from the talk with the design team it wasn't expected to have  `update/delete` operations, but I see no harm in having those ready